typescript/sdk/src/utils/fetch.ts

```diff
@@ -0,0 +1,14 @@
+import fetch from 'cross-fetch';
+
+// Adapted from https://github.com/node-fetch/node-fetch#request-cancellation-with-abortsignal
+export async function fetchWithTimeout(
```

Comment:
A slightly cleaner version that uses async/await correctly (this one need not use `async`)
https://github.com/hyperlane-xyz/hyperlane-explorer/blob/main/src/utils/timeout.ts#L28

Reply:
also added this recently :) https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/25dfcbaed7f50675b21f65507a6f84602da4af72/typescript/utils/src/utils.ts#L346

Reply:
Ahh nice. Just changed in favor of yours Rossy :)

Didn't use `runWithTimeout` because I have a feeling the signal triggers some cleanup that I'm not 100% sure would occur if I used runWithTimeout